web/client-api/src/main/java/io/deephaven/web/client/api/tree/JsTreeTable.java

```diff
@@ -726,19 +727,19 @@ private BitSet makeColumnSubscriptionBitset() {
     }
 
     @JsMethod
-    public void expand(Object row) {
-        setExpanded(row, true);
+    public void expand(Object row, @JsOptional Boolean expandAll) {
+        setExpanded(row, true, expandAll);
     }
 
     @JsMethod
     public void collapse(Object row) {
-        setExpanded(row, false);
+        setExpanded(row, false, false);
     }
 
     @JsMethod
-    public void setExpanded(Object row, boolean isExpanded) {
+    public void setExpanded(Object row, boolean isExpanded, @JsOptional Boolean expandAll) {
         // TODO check row number is within bounds
-
+        boolean expandAllValue = expandAll != null && expandAll == true;
```

Comment:
Can simplify null check and true check to just equals true - however in regular java at least you can't directly compare a null to a boolean primitive value (java will unbox it), instead you have to compare with the constant Boolean.TRUE.

This seems wasteful, but in GWT, since all your code is actually JS, all boxed booleans are just plain js values, that is in JS, `true === Boolean.TRUE`, there is no boxing and unboxing for Booleans (or Doubles).

So, this can be written simply as 
```suggestion
        boolean expandAllValue = expandAll == Boolean.TRUE;
```

Reply:
Then you could easily put this comparison into the actual "which expand byte should I send" check, rather than having two sets of checks, something like

```
final byte action;
if (!isExpanded) {
  // if you want to be picky, you could assert that expandAll is null or false here, to make sure no one is trying a "collapse with descendents"
  action = ACTION_COLLAPSE;
} else if (expandAll == Boolean.TRUE) {
  action = ACTION_EXPAND_WITH_DESCENDENTS;
} else {
  action = ACTION_EXPAND;
}
```